src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py

```diff
@@ -331,18 +335,27 @@ def recoverInjectedFiles(self):
         In that case run the recovery mode
         1. first check whether files which injection status = 0 are in the PhEDEx.
         2. if those file exist set the in_phedex status to 1
-        3. set self.filesToRecover = None
+        3. set self.blocksToRecover = None
+
+        Run this recovery one block at a time, with too many blocks
+        the call to the PhEDEx data service on cmsweb can time out
         """
         myThread = threading.currentThread()
-        
-        injectedFiles = self.phedex.getInjectedFiles(self.filesToRecover)
-        
-        myThread.transaction.begin()
-        self.setStatus.execute(injectedFiles, 1)
-        myThread.transaction.commit()
-        # when files are recovered set the self.file 
-        self.filesToRecover = None
-        return injectedFiles
+
+        # recover one block at a time
+        for block in self.blocksToRecover:
+
+            injectedFiles = self.phedex.getInjectedFiles(block)
```

Comment:
If this calls fails (due to some temporary network error), should we just log the error and keep doing for other block, although most likely subsequent call would fail in that case.